cli/main.c

```diff
@@ -137,6 +137,55 @@ static int list(int argc, char **argv)
 	return 0;
 }
 
+int print_dev_info(struct switchtec_dev *dev)
+{
+	int ret;
+	int device_id;
+	char *str;
+	char version[64];
+
+	device_id = switchtec_device_id(dev);
+
+	ret = switchtec_get_fw_version(dev, version, sizeof(version));
+	if (ret < 0) {
+		switchtec_perror("dev info");
+		return ret;
+	}
+
+	printf("%s:\n", switchtec_name(dev));
+	str =  switchtec_is_gen3(dev) ? "GEN3" :
```

Comment:
I'd suggest we need some static inline functions in switchtec.h to get the generation and variant strings.